### PR TITLE
Bael 7772 my batis vs hibernate

### PR DESCRIPTION
**Pull request by zeesh49**

### CODE REVIEW

File: persistence-modules/mybatis-vs-hibernate/pom.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        <dependency>
+            <groupId>org.mybatis</groupId>
+            <artifactId>mybatis</artifactId>
+            <version>3.5.16</version>
+        </dependency>
+        <dependency>
+        <groupId>org.hibernate.orm</groupId>
+        <artifactId>hibernate-core</artifactId>
+        <version>6.3.0.Final</version>
+        </dependency>
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/pom.xml:38]: Fix formatting + include a property like ${hibernate.version}

Same for the mybatis dependency

File: persistence-modules/mybatis-vs-hibernate/pom.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        <groupId>com.baeldung</groupId>
+        <artifactId>persistence-modules</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.10.2</version>
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/pom.xml:22]: Use existing ${junit-jupiter.version} from the parent

File: persistence-modules/mybatis-vs-hibernate/pom.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <version>2.2.224</version>
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/pom.xml:27]: Use existing ${h2.version} from the parent

File: persistence-modules/mybatis-vs-hibernate/pom.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        <dependency>
+        <groupId>org.hibernate.orm</groupId>
+        <artifactId>hibernate-core</artifactId>
+        <version>6.3.0.Final</version>
+        </dependency>
+    </dependencies>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/pom.xml:43]: These are not required, due to the way the parent modules are set up, specifying java.version is sufficient
```suggestion
```

File: persistence-modules/mybatis-vs-hibernate/src/main/java/com/baeldung/mybatisvshibernate/Student.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    private String name;
+
+    @Column
+    private int phone;
+
+    @Column
+    private String email;
+
+    public Student() {
+    }
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/src/main/java/com/baeldung/mybatisvshibernate/Student.java:26]: If this is required for Hibernate, make it clear with a comment
```suggestion
    public Student() {
        // required for Hibernate
    }
```

File: persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/HibernateUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    @Test
+    public void hibernateExample() {
+        assert sf != null;
+        Student zeesh = new Student("Zeeshan Arif", 246, "zeesh.arif@gmail.com");
+        Student bilal = new Student("Bilal Ahmed", 413, "xyz.123@gmail.com");
+        sf.inTransaction(session -> {
+            session.persist(zeesh);
+            session.persist(bilal);
+        });
+        ;
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/HibernateUnitTest.java:49]:
```suggestion

```

File: persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/HibernateUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        Student bilalFetched = sf.fromTransaction(session ->
+          session.find(Student.class, bilal.getId()));
+        Student zeeshFetched = sf.fromTransaction(session ->
+          session.find(Student.class, zeesh.getId()));
+
+        Assertions.assertEquals(zeesh, zeeshFetched);
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/HibernateUnitTest.java:55]: Add static import for assertEquals
```suggestion
        assertEquals(zeesh, zeeshFetched);
```
(similar elsewhere)

File: persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/MybatisUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    static SqlSessionFactory ssf;
+    @BeforeAll
+    public static void initialize() throws IOException {
+        String resource = "mybatis-config.xml";
+        InputStream inputStream = Resources.getResourceAsStream(resource);
+        ssf = new SqlSessionFactoryBuilder().build(inputStream);
+        myBatis3Setup();
+    }
+    private static void myBatis3Setup() throws IOException {
+        try(SqlSession sf = ssf.openSession()) {
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/MybatisUnitTest.java:27]: This whitespace issue is a sign that you might not be using the Baeldung formatter? Can you apply it to all files so we have our standard formatting rules applied
```suggestion
        try (SqlSession sf = ssf.openSession()) {
```

File: persistence-modules/mybatis-vs-hibernate/src/test/resources/StudentMapper.xml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+<mapper namespace="org.example.StudentMapper">
+    <select id="selectStudentById" parameterType="int" resultType="com.baeldung.mybatisvshibernate.Student">
+        select id, name, phone, email from dbMyBatis.Student where id = #{id}
+    </select>
+    <insert id="insertStudent" flushCache="true" keyColumn="id" keyProperty="id"  parameterType="com.baeldung.mybatisvshibernate.Student" useGeneratedKeys="true">
+        insert into dbMyBatis.Student (name,phone,email)
+        values (#{name},#{phone},#{email})
+        <!--<selectKey keyProperty = "id" resultType = "int" order = "AFTER">
+            SELECT CURRENT VALUE FOR dbMyBatis.STUDENT_TBL_SEQ
+        </selectKey>-->
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/src/test/resources/StudentMapper.xml:14]: Why is this commented out?

File: persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/MybatisUnitTest.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+            sr.runScript(new FileReader("src/test/resources/mybatis-create-db-and-table.sql"));
+        }
+    }
+
+    @Test
+    public void whenStudentsInsertedThenSearched() throws IOException {
+        Student zeesh = new Student("Zeeshan Arif", 246, "zeesh.arif@gmail.com");
+        Student bilal = new Student("Bilal Ahmed", 413, "xyz.123@gmail.com");
+        Student zeeshFetched;
+        Student bilalFetched;
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/MybatisUnitTest.java:38]: Move these declarations closer to their usage

File: persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/HibernateUnitTest.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+            .build();
+        try {
+            sf =
+              new MetadataSources(registry)
+                .addAnnotatedClass(Student.class)
+                .buildMetadata()
+                .buildSessionFactory();
+        } catch (Exception e) {
+            StandardServiceRegistryBuilder.destroy(registry);
+        }
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/HibernateUnitTest.java:36]: We want the test to fail if there is an exception here, so throw it:
```suggestion
        } catch (Exception e) {
            StandardServiceRegistryBuilder.destroy(registry);
            throw e;
        }
```

File: persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/HibernateUnitTest.java [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+    }
+
+    @Test
+    public void hibernateExample() {
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/HibernateUnitTest.java:41]: We use BDD style given_when_then or when_then naming for our test methods

File: persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/MybatisUnitTest.java [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    }
+    private static void myBatis3Setup() throws IOException {
+        try(SqlSession sf = ssf.openSession()) {
+            ScriptRunner sr = new ScriptRunner(sf.getConnection());
+            sr.runScript(new FileReader("src/test/resources/mybatis-create-db-and-table.sql"));
+        }
+    }
+
+    @Test
+    public void whenStudentsInsertedThenSearched() throws IOException {
```
Comment (theangrydev) [inline persistence-modules/mybatis-vs-hibernate/src/test/java/com/baeldung/mybatisvshibernate/MybatisUnitTest.java:34]: We use BDD style given_when_then or when_then naming for our test methods

**Review by theangrydev — CHANGES_REQUESTED** [review]

**Comment by lor6** [comment]
retest this please

**Pull request closed without merging**